components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -93,12 +98,13 @@ public void onMessage(Message message) {
             } else {
                 log.warn("Dropping the empty/null event received through jms receiver");
             }
-        } catch (JMSException e) {
+        } catch (JMSException | ArtifactSynchronizerException e) {
```

Comment:
Shall we handle this before?  in handleNotificationMessage method?

Reply:
Yes we are handling https://github.com/wso2/carbon-apimgt/pull/9019/files#diff-838be8956ccd306ac34942d1c83642fcR106

Reply:
Since this method is shared by lot of other features, better to capture the ArtifactSynchronizerException in previous method.